velox/exec/Aggregate.cpp

```diff
@@ -43,34 +43,56 @@ const AggregateFunctionEntry* FOLLY_NULLABLE
 getAggregateFunctionEntry(const std::string& name) {
   auto sanitizedName = sanitizeName(name);
 
-  auto& functionsMap = aggregateFunctions();
-  auto it = functionsMap.find(sanitizedName);
-  if (it != functionsMap.end()) {
-    return &it->second;
-  }
-  return nullptr;
+  return aggregateFunctions().withRLock(
```

Comment:
Hi @aditi-pandit, we recently had an internal user that accessed the simple scalar function registry concurrently with multiple threads and caused race condition (fixed in https://github.com/facebookincubator/velox/pull/4809). The vector scalar function registry has also been thread-safe. So I think it would be better to make the aggregation function registry thread-safe as well since we cannot control how Velox users access it.